content/draft/general/password_strength.md

```diff
@@ -0,0 +1,91 @@
+---
+title: "Password strength spec"
+description: ""
+date: 2020-10-06T08:48:57+00:00
+lastmod: 2020-10-06T08:48:57+00:00
+draft: false
+images: []
+menu:
+  raw:
+    parent: "general"
+weight: 100
+toc: true
+---
+
+# Password Strength spec
+
+## Motivation
+
+It's quite important the user chooses a strong password since the entire database is encrypted using that password. No matter how strong the encryption is, if the someone is able to brute force the password, they can read the database. This is why it's important to choose a strong password and to make sure it's not easy to guess. We can aid the user to choose a strong password by providing certain criteria.
+
+## Designs
+
+* [designs](https://www.figma.com/file/IPpvkpDWabBKJTeo6bFop0/Kuba%E2%8E%9CDesktop?node-id=11%3A4921)
+
+## Use Cases
+
+### Account creation
+
+1. User types the choosen password in the password field
+2. System displays feedback on the strength of the password
+2.1 System checks if the password fits criteria (see Functional Requirements#password criteria) and displays feedback (e.g "Password is too short")
+2.2 System categorizes password strength (see Functional Requirements#password categorization) and displays matching category (e.g password Is "Good")
+2.2 System checks if password is a common password (see Functional Requirements#password validation) and displays feedback (e.g "this pasword has been pwned and shouldn't be used")
+3. User types password again in the confirm password field
+4. System checks if password matches and displays "Passwords don't match" if they don't
+5. System checks if password matches and enables the "Create" button if they do
+6. User clicks "Create" button
+
+### Change password
+
+1. User types the current password in the current password field
+2. User types the choosen password in the new password field
+3. System displays feedback on the strength of the password
+4.1 System checks if the password fits criteria (see Functional Requirements#password criteria) and displays feedback (e.g "Password is too short")
+4.2 System categorizes password strength (see Functional Requirements#password categorization) and displays matching category (e.g password Is "Good")
+4.2 System checks if password is a common password (see Functional Requirements#password validation) and displays feedback (e.g "this pasword has been pwned and shouldn't be used")
+5. User types password again in the confirm password field
+6. System checks if password matches and displays "Passwords don't match" if they don't
+7. System checks if password matches and enables the "Create" button if they do
```

Comment:
There's no "Create" button in this scenario

Reply:
Correct, in this flow the button should be "Change Password"